WordPress/gutenberg
# Fix: Font size select value mismatch

Fixes: https://github.com/WordPress/gutenberg/issues/67283

## What?
The selected value gets updated in the font select after the next iteration. Thus leading to a lag in the correct value being displayed on the selector.

## Why?
The state in store.getState() is updated after the setValue logic is completed. This results in the currentValue being out of sync with the state during the render cycle. 

## How?
There is no need for the current value to be fetched from the store created when we have the current value available as a prop. This PR makes the change of using that value directly instead of waiting for the store's value to be set and then fetch it.

## Testing Instructions
1. In local set-up, Open `twentytwentyfive`'s `theme.json`. [ I have used this theme for demo, WP 6.8rc ].
2. In the fontSizes object ( settings -> typography -> fontSizes ), add the following objects at the end.
```
				{
					"fluid": {
						"max": "4rem",
						"min": "3.2rem"
					},
					"name": "Extra Extra Extra Large",
					"size": "3.2rem",
					"slug": "xxx-large"
				},
				{
					"fluid": {
						"max": "5rem",
						"min": "4.2rem"
					},
					"name": "Extra Extra Extra Extra Large",
					"size": "4.2rem",
					"slug": "xxxx-large"
				}
```
3. Open Global styles in site editor, and look for typography -> headings.
4. Check the font size select drop down displaying the correct values for all the different headings selected.

## Screenshots or screencast 
https://github.com/user-attachments/assets/7a529d62-06de-4f37-9e8a-9152c9708c2a
